setup.py

```diff
@@ -88,7 +88,7 @@ def find_version(*file_paths):
         "Source": "https://github.com/pykale/pykale",
     },
     packages=find_packages(exclude=("docs", "examples", "tests")),
-    python_requires=">=3.6",
+    python_requires=">=3.7, <=3.8",
```

Comment:
As it stands this will prevent installation on any python other than 3.7 or 3.8. @haipinglu should we keep 3.6 here and not in the tests?

Reply:
So far, only 3.6 can pass the tests so we can keep 3.6 until we make 3.7, 3.8, or 3.9 work. You may also think whether we should have a fresh new PR or continue in this PR.